source/promise-value.d.ts

```diff
@@ -20,8 +20,8 @@ type SyncData = PromiseValue<string>;
 let syncData: SyncData = getSyncData();
 
 // Here's an example that shows how this type reacts to recursive Promise types.
-type RecursiveAsyncData = Promise<Promise<string> >;
-let recursiveAsyncData: PromiseValue<RecursiveAsyncData> = Promise.resolve(Promise.resolve('ABC'));
+type RecursiveAsyncData = Promise<Promise<string>>;
+let recursiveAsyncData: PromiseValue<RecursiveAsyncData> = await Promise.resolve(Promise.resolve('ABC'));
```

Comment:
I don't think this is correct. The point is to show off how it handles nested promises.

Reply:
> The point is to show off how it handles nested promises.

Yup and how it handles nested promises is by unwrapping promises until the final resolved value.

![image](https://user-images.githubusercontent.com/32409546/163597357-f6595b83-e8a3-4a6c-96d2-c3014558635e.png)

`Promise.resolve()` is immediately returning a fulfilled promise with the result `'ABC'`, thus causing `recursiveAsyncData: PromiseValue<...>` to error `Type 'Promise<string>' is not assignable to type 'string'`
![image](https://user-images.githubusercontent.com/32409546/163597867-e7b2e35e-5344-4b7f-8693-12a957a40e61.png)

Perhaps, I'm missing something